pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -205,10 +205,16 @@ func (dswp *desiredStateOfWorldPopulator) findAndAddNewPods() {
 	}
 
 	for _, pod := range dswp.podManager.GetPods() {
+		if dswp.desiredStateOfWorld.GetHasReconstructionStarted() && dswp.podStateProvider.ShouldPodRuntimeBeRemoved(pod.UID) {
```

Comment:
I think we want to track the opposite: assume we're in the reconstruction phase, and then mark when reconstruction has completed.

Reply:
We want to add pod if it has started terminating at reconstruction?

And since we do not want to change:

```
if dswp.podStateProvider.ShouldPodContainersBeTerminating(pod.UID) {
      // Do not (re)add volumes for pods that can't also be starting containers
      continue
}
```
to

```
if dswp.podStateProvider.ShouldPodRuntimeBeRemoved(pod.UID) {
      // Do not (re)add volumes for pods that can't also be starting containers
      continue
}
```

We added the restriction to do the ShouldPodRuntimeBeRemoved check only if the reconstruction has started?

Reply:
What I mean is that when volume manager starts, we start in "reconstructing phase". Then once reconstruction is finished, we start processing volumes normally. So what I'm suggesting is that we assume we are reconstructing unless told otherwise.

But also, I think we don't actually need to depend on reconstruction state. I think we can just use the new check the first time the populator runs: 
https://github.com/kubernetes/kubernetes/blob/20b01afecbba85c53b58711b07049e8c1b764b2d/pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go#L158

@gnufied wdyt?

Reply:
Yeah, that should work. But I think this PR is not correct. Something like:

```golang
if dswp.hasAddedPods && dswp.podStateProvider.ShouldPodContainersBeTerminating(pod.UID) {
         continue
}

if !dswp.hasAddedPods && dswp.podStateProvider.ShouldPodRuntimeBeRemoved(pod.UID) {
    // Do not (re)add volumes for pods that can't also be starting containers
    continue
}

// otherwise add pods as expected.
```

Reply:
Also 6 months from now, we might forget why this behavior changes based on `dswp.hasAddedPods` , we will have to probably document the reasons..